### PR TITLE
Tweak: Added admin media modal hints [app-109]

### PR DESCRIPTION
**Pull request by bainternet**

### CODE REVIEW

File: core/admin/admin.php [hunk 1]
```diff
@@ -946,4 +950,44 @@ private function get_experiments() {
 	private function register_menu() {
 		$this->menus['main'] = new MainMenu();
 	}
+
+	private function maybe_enqueue_hints() {
+		if ( ! Hints::should_display_hint( 'image-optimization-once-media-modal' ) && ! Hints::should_display_hint( 'image-optimization-media-modal' ) ) {
+			return;
+		}
+		wp_register_script(
```
Comment (KingYes) [inline core/admin/admin.php:958]:
```suggestion

		wp_register_script(
```

File: assets/dev/js/admin/hints/media.js [hunk 2]
```diff
@@ -0,0 +1,110 @@
+( function() {
+	if ( ! elementorAdminHints || ! elementorAdminHints?.mediaHint ) {
+		return;
+	}
+	if ( ! wp || ! wp.media || ! wp.media.view || ! wp.media.view.Attachment || ! wp.media.view.Attachment.Details ) {
+		return;
+	}
+	wp.media.view.Attachment.Details = wp.media.view.Attachment.Details.extend( {
```
Comment (KingYes) [inline assets/dev/js/admin/hints/media.js:8]:
```suggestion

	wp.media.view.Attachment.Details = wp.media.view.Attachment.Details.extend( {
```

File: assets/dev/js/admin/hints/media.js [hunk 3]
```diff
@@ -0,0 +1,110 @@
+( function() {
+	if ( ! elementorAdminHints || ! elementorAdminHints?.mediaHint ) {
+		return;
+	}
+	if ( ! wp || ! wp.media || ! wp.media.view || ! wp.media.view.Attachment || ! wp.media.view.Attachment.Details ) {
```
Comment (KingYes) [inline assets/dev/js/admin/hints/media.js:5]:
```suggestion
	if ( ! wp.?media.?view.?Attachment.?Details ) {
```

File: assets/dev/js/admin/hints/media.js [hunk 4]
```diff
@@ -0,0 +1,110 @@
+( function() {
+	if ( ! elementorAdminHints || ! elementorAdminHints?.mediaHint ) {
```
Comment (KingYes) [inline assets/dev/js/admin/hints/media.js:2]:
```suggestion
	if ( ! elementorAdminHints?.mediaHint ) {
```

File: assets/dev/js/admin/hints/media.js [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+				</div>
+			</div>
+			<button class="elementor-control-notice-dismiss" data-event="<%= dismissible %>">
+				<i class="eicon eicon-close" aria-hidden="true"></i>
+				<span class="elementor-screen-only"><%= dismiss %></span>
+			</button>
+		</div>
+	</div>`,
+		template( view ) {
+		// Get the template and parse it
```
Comment (KingYes) [inline assets/dev/js/admin/hints/media.js:31]:
```suggestion
			// Get the template and parse it
```

**Review by KingYes — CHANGES_REQUESTED** [review]

File: assets/dev/js/admin/hints/media.js (hunk 3 above)
Comment (rami-elementor) [reply assets/dev/js/admin/hints/media.js:5]: @KingYes 
I think optional chaining is the other way `?.` not `.?`

File: assets/dev/js/admin/hints/media.js [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+			const settings = atob( b64Settings );
+			const { action_url: actionURL = null } = JSON.parse( settings );
+			if ( actionURL ) {
+				window.open( actionURL, '_blank' );
+			}
+			this.dismiss( event );
+		},
+
+		dismiss( event ) {
+			elementorCommon.ajax.addRequest( 'dismissed_editor_notices', {
```
Comment (rami-elementor) [inline assets/dev/js/admin/hints/media.js:97]: Not good, you are using the 'editor notices' mechanism to save data for the WordPress admin area.

Editor notices saves data with "widget name + control name".

See https://github.com/elementor/elementor/blob/main/includes/user.php, I believe there is a separate method for admin nitoces.

**Review by rami-elementor — DISMISSED** [review]

File: assets/dev/js/admin/hints/media.js (hunk 6 above)
Comment (bainternet) [reply assets/dev/js/admin/hints/media.js:98]: Nope, you are wrong, this is using the part of saving and retrieving the data but the same way the Notice control does. meaning the dismiss key is hardcoded and has no need for widget or control.

its already used for other hints in this same way.

File: core/admin/admin.php [hunk 7]
```diff
@@ -180,6 +183,7 @@ public function enqueue_styles() {
 		// It's for upgrade notice.
 		// TODO: enqueue this just if needed.
 		add_thickbox();
+
```
Comment (KingYes) [inline core/admin/admin.php:186]: @bainternet Why this is?

Comment (bainternet) [reply core/admin/admin.php:186]: the space?

Comment (bainternet) [reply core/admin/admin.php]: removed!

**Review by KingYes — APPROVED** [review]

**Review by arielk — APPROVED** [review]

**Pull request merged**